18 Normalizr/src/pages/student-list/actions/fetchCountryList.ts

```diff
@@ -0,0 +1,28 @@
+import { normalize } from 'normalizr';
+import { actionsEnums } from '../../../common/actionsEnums';
+import { countryApi } from '../../../rest-api/country-api';
+import {arrayOfCountriesSchema} from '../../../schemas/countrySchema';
+import {CountryView} from '../../../model/view/countryView';
+
+export const fetchCountryListRequestStartedAction = () => (dispatcher) => {
+  const promise = countryApi.fetchCountries();dispatcher
+
+  promise.then(
+    data => {
+      const countries = addInvalidCountryEntry(data);
+      const normalizedCountries = normalize(countries, arrayOfCountriesSchema);
+      dispatcher(fetchCountryListRequestCompletedAction(normalizedCountries));
+    }
+  );
+
+  return promise;
+}
+
+const addInvalidCountryEntry = (countries: CountryView[]) => {
+  return [{ id: 0, name: "Select country..." }, ...countries]
```

Comment:
This should be a UI-only concern, not something expected to be found in API/domain related code.